### PR TITLE
fix crash, when no pattern matches

### PR DESCRIPTION
**Pull request by dirkhub**
When no file matches the pattern, "value" is None. Comparisons between None-Type and integers are not allowed in newer Python versions and the "fileinfo_groups" check would crash. (Fix was tested in a 2.0.0p27cee site, but same problem still exists in the recent master branch.)

### CODE REVIEW

**Comment by kain88-de** [comment]
Thanks for fixing this bug. Can you also add a regression test in [test_fileinfo_utils.py](https://github.com/tribe29/checkmk/blob/master/tests/unit/cmk/base/plugins/agent_based/utils/test_fileinfo_utils.py), to ensure we do not accidentally introduce the bug again.

**Comment by kain88-de** [comment]
The Result checks during construction that it is not empty , either has a summary or a notice. To get rid of the error you should not yield the result. Unfortunately I'm sick and cannot look at the code currently. I'll send you an example once I'm better.

**Comment by kain88-de** [comment]
Hm I just had a look at the code. Without seeing your test I'm not sure where you run into the problem with `Result`. Could you add what you currently have as a commit. That would make it easier for me.

**Comment by dirkhub** [comment]
I added the commit for the additional unit tests. See lines 449 - 466 (and 374 - 398)

**Comment by dirkhub** [comment]
Hi, again! The main issue was, that I needed a unit test to check that no TypeError exception occurs. I did some further research and now found a suitable solution that catches the bug and passes, when the but is fixed. Please verify!

File: tests/unit/cmk/base/plugins/agent_based/utils/test_fileinfo_utils.py [hunk 1]
```diff
@@ -420,11 +443,36 @@ def test__fileinfo_check_function(  # type:ignore[no-untyped-def]
...
+        ),
+    ],
+)
+def test_typeerror__fileinfo_check_conjunctions(  # type:ignore[no-untyped-def]
+    check_definition, params
+) -> None:
+    try:
+        _result = list(_fileinfo_check_conjunctions(check_definition, params))
+    except TypeError:
+        pytest.fail("TypeError in _fileinfo_check_conjunctions with empty file list.")
```
Comment (kain88-de) [inline tests/unit/cmk/base/plugins/agent_based/utils/test_fileinfo_utils.py:478]: pytest has some handy helpers for this
```python
with pytest.raises(TypeError):
    list(_fileinfo_check_conjuncations(check_definition, params))
```

**Review by kain88-de — APPROVED** [review]
Looks good. I have to verify the tests and want to minimize them. 

I will make some small changes before merging this. You can look at the changes when github links the commit.

File: tests/unit/cmk/base/plugins/agent_based/utils/test_fileinfo_utils.py (hunk 1 above)
Comment (dirkhub) [reply tests/unit/cmk/base/plugins/agent_based/utils/test_fileinfo_utils.py:478]: with pytest.raises(TypeError)
expects a type error, right? But the unit test should pass, when NO TypeError is raised. I only found the solution with try and catch for this, as there ist no "pytest.notraises(TypeError)"

Comment (kain88-de) [reply tests/unit/cmk/base/plugins/agent_based/utils/test_fileinfo_utils.py:478]: So I now played around a bit with this. For cases where I do not want an exception to be thrown I usually do not use try/except. I just run the code in the try block and let any exception fail the test.

**Review by kain88-de — APPROVED** [review]
I finally understand your question with the empty result `expected=[Result(summary=None, notice=None)]`. This is now allowed with out type system. The solution is to use no result `expected=[]`

I will merge this soon with some small changes. Thank you for your contribution.

**Pull request closed without merging**